src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -555,11 +555,22 @@ export class ResultsViewPageStore {
             : [];
     }
 
-    @computed get treatmentList() {
-        return this.urlWrapper.query.treatment_list &&
-            this.urlWrapper.query.treatment_list.trim().length
-            ? this.urlWrapper.query.treatment_list.trim().split(/;/)
+    @computed get parsedGenericAssayGroups() {
+        // empty if no heatmap tracks param
+        const groups = this.urlWrapper.query.generic_assay_groups
+            ? this.urlWrapper.query.generic_assay_groups
+                  .split(';')
+                  .map((x: string) => x.split(','))
             : [];
+
+        const parsedGroups = groups.reduce(
+            (acc: { [molecularProfileId: string]: string[] }, group) => {
+                acc[group[0] as string] = group.slice(1);
```

Comment:
why not use group[1] ? instead of slice?  just curious

Reply:
because every group is a string like this `molecular_profile_id,entity_1,entity_2`, the first one in the array stores the value of `molecular_profile_id`, group[1] will just take the `entity_1`, we want the whole array except the first `molecular_profile_id`. That's why I use `group.slice(1)` at here.